.github/workflows/deploy_SNAPSHOT.yml

```diff
@@ -10,11 +10,12 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
 
     - name: Set up JDK 17
-      uses: actions/setup-java@v1
+      uses: actions/setup-java@v3
       with:
+        distribution: 'temurin'
```

Comment:
@chengenzhao Thanks for this. Looks good, just one question: why do we need to specify `distribution`

Reply:
Because this field is required when upgrading to Version 3
it could be ignored for v1&2, but is required when using v3 
<img width="363" alt="截屏2023-01-04 上午11 17 33" src="https://user-images.githubusercontent.com/5525436/210477936-b4c90d28-a387-4a7a-9f67-7fd5fe672847.png">
https://github.com/actions/setup-java

Reply:
There are other options though
Oracle official provide its own JDK setup
e.g.
```github action
         uses: oracle-actions/setup-java@v1
         with:
           website: jdk.java.net
           release: 19
```
and Graal also provide JDK setup
e.g.
```github action
        uses: graalvm/setup-graalvm@v1
        with:
          version: 'latest'
          java-version: '19'
```
Both works

Reply:
Great thanks